src/big_query/big_query.py

```diff
@@ -72,7 +72,16 @@ def list_table_ids(self, project_id, dataset_id):
             projectId=project_id, datasetId=dataset_id
         )
         while request is not None:
-            tables = request.execute()
+            try:
+                tables = request.execute()
+            except HttpError as ex:
+                if ex.resp.status == 404:
```

Comment:
Does service return 404 only for missing dataset?

Reply:
no. I added additional validation for that situation